R/projpred.R

```diff
@@ -121,11 +121,23 @@ get_refmodel.brmsfit <- function(object, newdata = NULL, resp = NULL,
     .extract_model_data(object, newdata = newdata, resp = resp, ...)
   }
   
+  # extract a list of K-fold sub-models
+  if (!inherits(cvfun, "function")) {
+    cvfun <- function(folds) {
+      cvres <- kfold(
+        object, K = max(folds),
+        save_fits = TRUE, folds = folds
+      )
+      fits <- cvres$fits[, "fit"]
+      return(fits)
+    }
+  }
+  
```

Comment:
Does `brms::kfold` automatically check `options(mc.cores)` for parallelization? Anyway, we want to have control over that from projpred at least, so that passing `cvfun(..., cores = N)` uses only `N` cores.

Reply:
9c21a6340abd6eb7c68e4fe2b06977661761494c should allow `cvfun` to pass extra arguments to `kfold`. I'll defer to @paul-buerkner to make sure that `kfold` accepts `cores` argument and use it if not.

On a related note, the checks in `projpred` can be generic. I'll update https://github.com/stan-dev/projpred/pull/116.